group.go

```diff
@@ -152,3 +153,50 @@ func (s *GroupService) Remove(groupname string, username string) (*Response, err
 
 	return resp, nil
 }
+
+// Create creates an group in JIRA.
+//
+// JIRA API docs: https://developer.atlassian.com/cloud/jira/platform/rest/#api-api-2-group-post
+func (s *GroupService) Create(group *Group) (*Group, *Response, error) {
+	apiEndpoint := "/rest/api/2/group"
+	req, err := s.client.NewRequest("POST", apiEndpoint, group)
+	if err != nil {
+		return nil, nil, err
+	}
+
+	resp, err := s.client.Do(req, nil)
+	if err != nil {
+		return nil, resp, err
+	}
+
+	responseGroup := new(Group)
+	defer resp.Body.Close()
```

Comment:
You don't need to do all of this.  The `Do` function already handles deserializing the JSON into the struct.